File: Proyectos móviles/AppExamenAxel/app/src/main/java/una/acr/appexamenaxel/view/activities/MainActivity.java
package una.acr.appexamenaxel.view.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.airbnb.lottie.LottieAnimationView;

import java.util.ArrayList;

import es.dmoral.toasty.Toasty;
import una.acr.appexamenaxel.R;
import una.acr.appexamenaxel.domain.Cliente;
import una.acr.appexamenaxel.data.ClienteData;
import una.acr.appexamenaxel.view.adapters.ClienteAdapter;

public class MainActivity extends AppCompatActivity {

    private ClienteData clienteData;
    private RecyclerView recyclerClientes;
    private ClienteAdapter clienteAdapter;
    private ArrayList<Cliente> listaClientes;

    private Button btnAgregarCliente;
    private Button btnAgregarProducto, btnVincularProducto;

    private LottieAnimationView clientes, productos, favoritos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        clienteData = new ClienteData(this);
        setContentView(R.layout.activity_main);
        clientes = findViewById(R.id.btn_agregar_cliente);
        productos = findViewById(R.id.btn_agregar_producto);
        favoritos = findViewById(R.id.btn_vincular_producto);



        listaClientes = new ArrayList<>();
        clienteAdapter = new ClienteAdapter(listaClientes);
        recyclerClientes = findViewById(R.id.recycler_clientes);
        recyclerClientes.setLayoutManager(new LinearLayoutManager(this,RecyclerView.VERTICAL,false));
        recyclerClientes.setAdapter(clienteAdapter);
        actualizarLista();


        productos.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,FormularioProductoActivity.class);
                startActivity(intent);
                finish();
            }
        });

        clientes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,FormularioClienteActivity.class);
                intent.putExtra("tipo","insertar");
                startActivity(intent);
                finish();
            }
        });

        favoritos.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, VincularFavoritosActivity.class);
                startActivity(intent);
                finish();
            }
        });

        if(clienteAdapter.getItemCount()<=0){
            Toasty.info(getApplicationContext(), "No existen clientes registrados", Toast.LENGTH_LONG, true).show();
            //Toast.makeText(getApplicationContext(),"No existen clientes registrados",Toast.LENGTH_SHORT).show();
        }

        //clienteData.insertarCliente(new Cliente("Test",88888888,0));
    }

    private void actualizarLista() {
        Cursor cursor = clienteData.obtenerClientes();
        listaClientes.clear();
        if(clienteAdapter == null){
            return;
        }else{
            if(cursor == null){
                return ;
            }else{
                while(cursor.moveToNext()){
                    listaClientes.add(new Cliente(cursor.getInt(0),cursor.getString(1),cursor.getInt(2),cursor.getInt(3)));
                }
            }

            cursor.close();

        }

        clienteAdapter.setLista(listaClientes);
        clienteAdapter.notifyDataSetChanged();
    }
}

File: Proyectos móviles/kotlin/UsuariosApp/app/src/main/java/una/acr/usuariosapp/views/activities/MainActivity.kt
package una.acr.usuariosapp.views.activities

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.airbnb.lottie.LottieAnimationView
import una.ac.cr.usuariosapp.domain.Usuario
import una.acr.usuariosapp.R
import una.acr.usuariosapp.network.VolleyUsuario
import una.acr.usuariosapp.views.adapters.UsuarioAdapter
import una.acr.usuariosapp.views.interfaces.UsuarioDeleteCallback
import una.acr.usuariosapp.views.interfaces.UsuarioICallback
import java.util.jar.Attributes

class MainActivity : AppCompatActivity() , UsuarioDeleteCallback {
    private lateinit var btnAgregar : LottieAnimationView;
    private lateinit var recyclerUsuarios : RecyclerView;
    private var listaUsuarios: ArrayList<Usuario> = ArrayList()
    private lateinit var usuarioAdapter: UsuarioAdapter


    internal val launcher =
        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
            if (result.resultCode == Activity.RESULT_OK) {
                Toast.makeText(this, "Agregado con éxito", Toast.LENGTH_LONG).show()
               actualizarLista()
            } else if (result.resultCode == 200) {

                Toast.makeText(this, "Actualizado con éxito", Toast.LENGTH_LONG).show()
                actualizarLista()
            } else if (result.resultCode == 400) {

                Toast.makeText(this, "Error al actualizar", Toast.LENGTH_LONG).show()

            }
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        WindowCompat.setDecorFitsSystemWindows(window, false)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

       // listaUsuarios.add(Usuario(1,"Axel","siuu","22","33","2",1))




        btnAgregar = findViewById(R.id.btn_agregar_usuario)


        btnAgregar.setOnClickListener {
            val intent = Intent(this, FormularioUsuarioActivity::class.java)
            intent.putExtra("metodo", "agregar")
            launcher.launch(intent)
        }

        recyclerUsuarios = findViewById(R.id.recyclerv_usuarios)
        recyclerUsuarios.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)

        val volleyUsuario = VolleyUsuario()
        val listener = object : UsuarioICallback {
            override fun onUsuarioReceived(lista: ArrayList<Usuario>) {
                listaUsuarios = lista
                usuarioAdapter.setLista(listaUsuarios)


            }
        }

        volleyUsuario.obtenerUsuarios(
            this,
            "",
            listener
        )

        usuarioAdapter = UsuarioAdapter( listaUsuarios, launcher, applicationContext,this)
        recyclerUsuarios.adapter = usuarioAdapter
    }

    private fun actualizarLista() {
        val volleyUsuario = VolleyUsuario()

        val listener = object : UsuarioICallback {
            override fun onUsuarioReceived(lista: ArrayList<Usuario>) {
                listaUsuarios.clear()
                listaUsuarios.addAll(lista)
                usuarioAdapter.notifyDataSetChanged()
            }
        }

        volleyUsuario.obtenerUsuarios(this, "", listener)
    }

    override fun onUsuarioDeleted() {
        actualizarLista()
    }

}